src/main/java/ch/njol/skript/expressions/ExprSignText.java

```diff
@@ -66,84 +66,89 @@ public class ExprSignText extends SimpleExpression<String> {
 	@SuppressWarnings("null")
 	private Expression<Number> line;
 	@SuppressWarnings("null")
-	private Expression<Block> exprBlock;
+	private Expression<Block> block;
 	private boolean delayed;
 
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		isLines = (matchedPattern == 0);
-		if (!isLines)
-			line = matchedPattern == 1 ? (Expression<Number>) exprs[0] : new SimpleLiteral<>(parseResult.mark, false);
-		exprBlock = (Expression<Block>) exprs[exprs.length - 1];
+		if (!isLines) {
+			if (matchedPattern == 1) {
+				line = (Expression<Number>) exprs[0];
+			} else {
+				new SimpleLiteral<>(parseResult.mark, false);
+			}
```

Comment:
you call `new SimpleLiteral` here but never set `line` to this

Reply:
Oops.

Reply:
I merged this with ternary but LimeGlass disagreed, so I reverted back to this.

Reply:
alright